packages/template-chakra-storefront/src/pages/cart/hooks/use-cart-operations.js

```diff
@@ -39,6 +43,8 @@ export const useCartOperations = (basket, productsByItemId, showError) => {
         // using try-catch is better than using onError callback since we have many mutation calls logic here
         try {
             setCartItemLoading(true)
+            // close the modal before performing any actions on cart item
+            onClose()
```

Comment:
The root cause of the bug is that the modal is not close when clicking "update" button from product view. We need to move the modal from cart page up to cartOperation hook to be able to do that. This will resolve the bug